orttraining/orttraining/core/framework/torch/custom_function_register.cc

```diff
@@ -88,11 +88,14 @@ void OrtTorchFunctionPool::RegisterTorchAutogradFunction(
   PythonObjectPtr forward(PyObject_GetAttrString(obj, "apply"), PythonObjectDeleter);
   PythonObjectPtr backward(PyObject_GetAttrString(obj, "backward"), PythonObjectDeleter);
 
+  PythonObjectPtr unsafe_forward(PyObject_GetAttrString(obj, "forward"), PythonObjectDeleter);
   ORT_ENFORCE(forward.get(), "apply attribute not found when registering ", key);
   ORT_ENFORCE(backward.get(), "backward attribute not found when registering ", key);
+  ORT_ENFORCE(unsafe_forward.get(), "forward attribute not found when registering ", key);
```

Comment:
What is `unsafe_forward`?

Reply:
As we chatted offline, it means, ctx will not be created by Torch, and we will not call THPFunction_apply to call the forward, instead, we call the static forward function as a normal function. This mode is used by those autograd functions that are not use fancy APIs of torch ctx, for example, ctx.save_for_backward, etc. It can save the xx us overhead in  THPFunction_apply before and after the real function execution.